app/javascript/orangelight/availability.es6

```diff
@@ -115,6 +116,32 @@ export default class AvailabilityUpdater {
 
   // show page
   process_single(holding_records) {
+    var dataComplete = true;
+    for (let holding_id in holding_records[this.id]) {
+      const availability_info = holding_records[this.id][holding_id];
+      if (availability_info['temp_location'] === true) {
+        dataComplete = false; // The data that we get from Alma for temporary locations is incomplete.
+        break;
+      }
+    }
+
+    if (dataComplete) {
+      // Update the page with the data that we already have.
+      this.update_single(holding_records);
+      return;
+    }
+
+    // Make a separate call (with deep=true) to get more information before updating the page.
+    var url = `${this.bibdata_base_url}/bibliographic/${this.id}/availability.json?deep=true`;
+    $.getJSON(url, this.update_single)
```

Comment:
Makes second call to bib data and uses the result of this call to update the page.

Reply:
The incomplete data looks like this (notice the `fake_id`):

```
{
    "9959958323506421": {
        "fake_id_1": {
            "on_reserve": "N",
            "location": "lewis$resterm",
            "label": "Lewis Library: Term Loan Reserves",
            "status_label": "Available",
            "copy_number": null,
            "cdl": false,
            "temp_location": true,
            "id": "fake_id_1"
        }
    }
}
```

The second call will give the return the proper data (notice the correct holding_ids):
```
{
    "9959958323506421": {
        "22272063570006421": {
            "on_reserve": "N",
            "location": "lewis$resterm",
            "label": "sciresp: Lewis: Term Loan",
            "status_label": "Available",
            "copy_number": "1",
            "cdl": false,
            "temp_location": true,
            "id": "22272063570006421"
        },
        "22272063520006421": {
            "on_reserve": "N",
            "location": "lewis$resterm",
            "label": "sciresp: Lewis: Term Loan",
            "status_label": "Available",
            "copy_number": "2",
            "cdl": false,
            "temp_location": true,
            "id": "22272063520006421"
        }
    }
}
```